pkg/e2e/addons/addon_test_harness.go

```diff
@@ -24,6 +26,11 @@ var _ = ginkgo.Describe("[Suite: addons] Addon Test Harness", func() {
 	}
 	ginkgo.It("should run until completion", func() {
 		h.SetServiceAccount(viper.GetString(config.Addons.TestUser))
-		h.RunAddonTests("addon-tests", int(addonTimeoutInSeconds), strings.Split(viper.GetString(config.Addons.TestHarnesses), ","), []string{})
+		harnesses := strings.Split(viper.GetString(config.Addons.TestHarnesses), ",")
+		failed := h.RunAddonTests("addon-tests", int(addonTimeoutInSeconds), harnesses, []string{})
+		if len(failed) > 0 {
+			// tests failed, notify
+			alert.SendSlackMessage(viper.GetString(config.Addons.SlackChannel), fmt.Sprintf("Addon tests failed: %v", failed))
```

Comment:
Can you either wrap this in a check to see if the Slack API token is set (or do it as part of SendSlackMessage)?

Currently for normal addon jobs we don't set it so once this is in we'll have to cut a PR against openshift/release to add the Slack token into it. :)